drivers/src/bounded_address.rs

```diff
@@ -24,18 +24,19 @@ impl MemBounds for RomBounds {
 pub type RomAddr<T> = BoundedAddr<T, RomBounds>;
 
 #[repr(C)]
-#[derive(Zeroize)]
-pub struct BoundedAddr<T: AsBytes + FromBytes, B: MemBounds> {
+#[derive(Zeroize, TryFromBytes, Immutable)]
+pub struct BoundedAddr<T: IntoBytes + FromBytes, B: MemBounds> {
     addr: u32,
     _phantom: PhantomData<(T, B)>,
 }
-unsafe impl<T: AsBytes + FromBytes, B: MemBounds> FromBytes for BoundedAddr<T, B> {
+// Unaligned is not implemented for `u32`, so `BoundedAddr` cannot currently derive `IntoBytes`.
+// Potentially `u32` can be swapped with `U32` from the zerocopy crate, but this type does not
+// implement Zeroize.
+// We could potentially wrap `U32` with the `Zeroizing` type to resolve this.
```

Comment:
What's the conclusion here? It seems like it's maybe working now.

Reply:
I haven't followed up on this, but for this case I think we will not be able to use the derive feature.

I believe the orphan rule will block the workaround (Cannot implement `Zeroizing` (zeroize trait) on `U32` (Unaligned u32 struct from zerocopy).

Reply:
Looked at this again with fresh eyes and cleaned it up. This type can now just derive the zerocopy traits directly.